src/FluentMigrator.Runner/MigrationRunner.cs

```diff
@@ -280,7 +259,6 @@ public virtual void ApplyMigrationUp(IMigrationInfo migrationInfo, bool useTrans
                 _alreadyOutputPreviewOnlyModeWarning = true;
             }
 
-            if (!migrationInfo.IsAttributed() || !VersionLoader.VersionInfo.HasAppliedMigration(migrationInfo.Version))
```

Comment:
Due to the fact that this function is public, the removal of this condition would be a change in behaviour. When you revert the removal of this code, then I'll happily apply your PR.